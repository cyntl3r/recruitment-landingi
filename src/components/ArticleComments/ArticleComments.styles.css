.list {
  list-style: none;
  margin: 0;
  padding: 0;
}

.listItem {
  font-size: 1rem;
  margin-top: 30px;
  background-color: #f0f0f0;
  padding: 10px;
}

.listItem p {
  margin: 5px 0;
}
