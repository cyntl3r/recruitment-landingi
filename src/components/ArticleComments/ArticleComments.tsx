import { FunctionComponent, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { addNewComment, fetchCommentsOfArticle } from '../../actions/comment';
import { CommentState } from '../../reducers/commentReducer';
import { AppState } from '../../store';
import { ArticleCommentsAdd } from '../ArticleCommentsAdd/ArticleCommentsAdd';
import styles from './ArticleComments.styles.css';

export interface ArticleCommentsProps {
  articleId: number;
}

export const ArticleComments: FunctionComponent<ArticleCommentsProps> = ({
  articleId,
}) => {
  const dispatch = useDispatch();
  const { loading, comments } = useSelector<AppState, CommentState>(
    (state) => state.comment
  );

  const handleOnNewCommentSubmit = (email: string, body: string) => {
    dispatch(addNewComment(articleId, email, body));
  };

  useEffect(() => {
    dispatch(fetchCommentsOfArticle(articleId));
  }, []);

  return (
    <div data-testid="article-comments">
      <ArticleCommentsAdd onSubmit={handleOnNewCommentSubmit} />
      <h4>Comments</h4>
      {loading ? (
        'Loading...'
      ) : !comments.length ? (
        <p>no comments</p>
      ) : (
        <ul className={styles.list}>
          {comments.map(({ id, email, body }) => (
            <li key={id} className={styles.listItem}>
              <b>- {email}</b>
              <p>{body}</p>
            </li>
          ))}
        </ul>
      )}
    </div>
  );
};
