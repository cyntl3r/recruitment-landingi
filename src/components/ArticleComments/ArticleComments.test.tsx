import { render } from '../../testing/test-utils';
import { ArticleComments } from './ArticleComments';

test('snapshot', () => {
  const { container } = render(<ArticleComments articleId={1} />);
  expect(container).toMatchSnapshot();
});

test('wrapper render', () => {
  const { getByTestId } = render(<ArticleComments articleId={10} />);
  expect(getByTestId('article-comments')).toBeDefined();
});
