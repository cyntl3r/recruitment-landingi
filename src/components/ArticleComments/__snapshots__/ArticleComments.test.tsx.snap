// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <div
    data-testid="article-comments"
  >
    <div
      data-testid="article-comments-add"
    >
      <h4>
        Add comment
      </h4>
      <form>
        <textarea
          placeholder="Comment..."
          required=""
        />
        <div>
          <input
            placeholder="E-mail address"
            required=""
            type="email"
            value=""
          />
          <button
            type="submit"
          >
            Add
          </button>
        </div>
      </form>
    </div>
    <h4>
      Comments
    </h4>
    Loading...
  </div>
</div>
`;
